pontoon/base/models.py

```diff
@@ -850,6 +850,83 @@ def __unicode__(self):
         return self.name
 
 
+class EntityQuerySet(models.QuerySet):
+    """
+    Queryset provides a set of additional methods that should allow us to filter entities
+    in the same way as the frontend code does.
+    Methods should cover
+    """
+    def with_status_counts(self, locale):
+        """
+        Helper method that returns an set with annotation of following fields:
+            * approved_count - a number of approved translations in the entity.
+            * fuzzy_count - a number of fuzzy translations in the antity
+            * translations_count - a number of translations assigned do the entity.
+            * expected_count - a number of translations that should cover entity.
+            * unchanged_count - a number of translations that has the same string as the entity.
+        """
+        return self.annotate(
+            approved_count=Sum(
+                Case(
+                    When(
+                        Q(translation__approved=True, translation__locale=locale), then=1
+                    ), output_field=models.IntegerField(), default=0
+                )
+            ),
+            fuzzy_count=Sum(
+                Case(
+                    When(
+                        Q(translation__fuzzy=True, translation__locale=locale), then=1
+                    ), output_field=models.IntegerField(), default=0
+                )
+            ),
+            translations_count=Sum(
+                Case(
+                    When(
+                        Q(translation__locale=locale), then=1
+                    ), output_field=models.IntegerField(), default=0
```

Comment:
You also need to make sure translation is not approved or fuzzy.

I suggest renaming this to 'translated_count' to fit the rest of the code or even 'suggested_count' to match the expression we use in the UI.